src/mdast/tokensToMyst.ts

```diff
@@ -0,0 +1,506 @@
+import { Spec, Token, Container, Admonition, AdmonitionKind } from './types';
+import { visit } from 'unist-util-visit';
+import { remove } from 'unist-util-remove';
+import { u } from 'unist-builder';
+import he from 'he';
+import { GenericNode } from '.';
+import { admonitionKindToTitle, withoutTrailingNewline } from './utils';
+import { map } from 'unist-util-map';
+import { findAfter } from 'unist-util-find-after';
+
+export type Options = {
+  handlers?: Record<string, Spec>;
+  hoistSingleImagesOutofParagraphs?: boolean;
+  nestBlocks?: boolean;
+};
+
+function getClassName(token: Token, exclude?: RegExp): string | undefined {
+  const className: string = token.meta?.class?.join(' ') || token.attrGet('class');
+  if (!className) return undefined;
+  return (
+    className
+      .split(' ')
+      .map((c) => c.trim())
+      .filter((c) => c && !(exclude && c.match(exclude)))
+      .join(' ') || undefined
+  );
+}
+
+function hasClassName(token: Token, matcher: RegExp): false | RegExpMatchArray {
+  const className = getClassName(token);
+  if (!className) return false;
+  const matches = className
+    .split(' ')
+    .map((c) => c.match(matcher))
+    .filter((c) => c);
+  if (matches.length === 0) return false;
+  return matches[0] as RegExpMatchArray;
+}
+
+function getLang(t: Token) {
+  return he.decode(t.info).trim().split(' ')[0].replace('\\', '');
+}
+
+function normalizeLabel(label: string | undefined) {
+  return label?.replace(/[\s]+/g, ' ').trim().toLowerCase();
+}
+
+const defaultMdast: Record<string, Spec> = {
+  heading: {
+    type: 'heading',
+    getAttrs(token) {
+      return { depth: Number(token.tag[1]) };
+    },
+  },
+  hr: {
+    type: 'thematicBreak',
+    noCloseToken: true,
+    isLeaf: true,
+  },
+  paragraph: {
+    type: 'paragraph',
+  },
+  blockquote: {
+    type: 'blockquote',
+  },
+  ordered_list: {
+    type: 'list',
+    getAttrs(token, tokens, index) {
+      const info = tokens[index + 1]?.info;
+      const start = Number(tokens[index + 1]?.info);
+      return {
+        ordered: true,
+        start: isNaN(start) || !info ? 1 : start,
+        spread: false,
+      };
+    },
+  },
+  bullet_list: {
+    type: 'list',
+    attrs: {
+      ordered: false,
+      spread: false,
+    },
+  },
+  list_item: {
+    type: 'listItem',
+    attrs: {
+      spread: true,
+    },
+  },
+  em: {
+    type: 'emphasis',
+  },
+  strong: {
+    type: 'strong',
+  },
+  fence: {
+    type: 'code',
+    isLeaf: true,
+    getAttrs(t) {
+      return { lang: getLang(t), value: withoutTrailingNewline(t.content) };
+    },
+  },
+  code_block: {
+    type: 'code',
+    isLeaf: true,
+    getAttrs(t) {
+      return { lang: getLang(t), value: withoutTrailingNewline(t.content) };
+    },
+  },
+  code_inline: {
+    type: 'inlineCode',
+    noCloseToken: true,
+    isText: true,
+  },
+  hardbreak: {
+    type: 'break',
+    noCloseToken: true,
+    isLeaf: true,
+  },
+  link: {
```

Comment:
We need to make sure any links like `[](my-ref) are converted to `contentReferences` if `my-ref` is an existing identifier.

I think we can just do this in state resolution: visit all `links`, if `url in state.targets`, change the node to `contentReference`.